### DISCUSSION TITLE
Why does create and update methods from MedusaService class have any type

### DISCUSSION THREAD

**Discussion by samyung0** [medusajs/medusa#10239] [Q&A] [answered]
In [types/medusa-service.ts](https://github.com/medusajs/medusa/blob/617a5972bff2ce99056af362764d4b14a9c9feff/packages/core/utils/src/modules-sdk/types/medusa-service.ts), dev commented that there is a bug with the update and create typings and marked it as TODO, but so far no fixes have been made.

So the question is 1. what bugs are related and 2. how can we workaround this? If I return the value from the create and update methods in steps/workflow, the whole function has any return type which is bad

**Comment by samyung0** [comment]
@adrien2p Sorry for the mention but would it be fine for you to shed some light?

**Reply by adrien2p** [reply, marked as answer]
Hey @samyung0,

Thank you for asking, 
The main reason is that it needs to be reworked but as not been prioritised yet (we have a lot of work and we are still a small team).

Other than that, the other reasons are:

- if we use a partial type of T then when overriding if you want something else then you cant because an override must respect the api of the method being overridden. As a solution you could still ignore the type error but it still create the other way around issue/feeling.

- instance method mapped types are not yet possible leading to being recognised as instance member properties forcing the override to ignore the error and therefore hide potential type issues.

So, to summarise 😅
We are aware of this part and that it needs to be reworked a bit, there are still some language limitations and decisions that we need to make, and finally it will be prioritised internally at some point depending on many other parameters.

Pr are still welcomed 🙏

**Reply by samyung0** [reply]
Appreciate the prompt response. I would love to contribute in my free time if that's possible :)

I just started out a couple days ago and I am still reading through the docs, but imo you guys have already done an amazing job so far with the architectural design and all the customization and I really like it. I will dig deeper into the codes once I have gone through all the guides.

**Reply by adrien2p** [reply]
I couldn't be more happy when I ear that 🥰 but I really appreciate your feedback and I am thankful for your kind words.

Hope to see you soon in the contributor list and thank you for your understanding 😉
